middleware/parse/parse_test.go

```diff
@@ -100,13 +101,20 @@ func TestParseMiddleware(t *testing.T) {
 
 	t.Run("JSON Invalid", func(t *testing.T) {
 		request := testutil.NewTestRequest(http.MethodPost, "/parse", bytes.NewBuffer([]byte("{\"unclosed\"")))
+		request.Lang = server.Lang.GetDefault()
 		request.Header().Set("Content-Type", "application/json")
 
 		result := server.TestMiddleware(&Middleware{MaxUploadSize: 0.01}, request, func(_ *goyave.Response, _ *goyave.Request) {
 			assert.Fail(t, "Middleware should not pass")
 		})
 		assert.NoError(t, result.Body.Close())
 		assert.Equal(t, http.StatusBadRequest, result.StatusCode)
+		assert.NotNil(t, request.Extra[goyave.ExtraParseError{}])
+		assert.NotPanics(t, func() {
+			extraError, ok := request.Extra[goyave.ExtraParseError{}].(error)
```

Comment:
Your type assertion  is no longer needed


```suggestion
			assert.Error(the, request.Extra[goyave.ExtraParseError{}])
```


You should use


```suggestion
			require.Contains(t, request.Extra,goyave.ExtraParseError{})
     assert.Error(the, request.Extra[goyave.ExtraParseError{}])
```

https://pkg.go.dev/github.com/stretchr/testify/require#Contains

Then the notpanic is no longer needed I think

Reply:
Can't assert an error, if I don't type it as error.

Reply:
Indeed 🤦‍♀️